pkg/vcs/github_client/client.go

```diff
@@ -156,15 +171,62 @@ func (c *Client) buildRepoFromEvent(event *github.PullRequestEvent) vcs.PullRequ
 	}
 
 	return vcs.PullRequest{
-		BaseRef:       *event.PullRequest.Base.Ref,
-		HeadRef:       *event.PullRequest.Head.Ref,
-		DefaultBranch: *event.Repo.DefaultBranch,
-		CloneURL:      *event.Repo.CloneURL,
+		BaseRef:       event.PullRequest.Base.GetRef(),
+		HeadRef:       event.PullRequest.Head.GetRef(),
+		DefaultBranch: event.Repo.GetDefaultBranch(),
+		CloneURL:      event.Repo.GetCloneURL(),
 		FullName:      event.Repo.GetFullName(),
-		Owner:         *event.Repo.Owner.Login,
+		Owner:         event.Repo.Owner.GetLogin(),
 		Name:          event.Repo.GetName(),
-		CheckID:       *event.PullRequest.Number,
-		SHA:           *event.PullRequest.Head.SHA,
+		CheckID:       event.PullRequest.GetNumber(),
+		SHA:           event.PullRequest.Head.GetSHA(),
+		Username:      c.username,
+		Email:         c.email,
+		Labels:        labels,
+
+		Config: c.cfg,
+	}
+}
+
+// buildRepoFromComment builds a vcs.PullRequest from a github.IssueCommentEvent
+func (c *Client) buildRepoFromComment(context context.Context, comment *github.IssueCommentEvent) vcs.PullRequest {
+	prURL, err := url.Parse(comment.Issue.GetURL())
+	if err != nil {
+		log.Error().Msgf("failed to parse pr url: %+v", prURL)
+		return nilPr
+	}
+	pathSegments := strings.Split(prURL.Path, "/")
+	if len(pathSegments) < 5 {
+		// invalid path, return nilPr
+		log.Error().Msgf("invalid pr url path: %s", prURL.Path)
+		return nilPr
+	}
+	owner := pathSegments[2]
+	repo := pathSegments[3]
+	prNumber, err := strconv.Atoi(pathSegments[4])
```

Comment:
Could we get rid of the string parsing? 

```suggestion
	owner := comment.GetIssue().GetRepository().GetOwner().GetName()
	repo := comment.GetIssue().GetRepository().GetName()
	prNumber := comment.GetIssue().GetNumber()
```

Reply:
sounds ok to me